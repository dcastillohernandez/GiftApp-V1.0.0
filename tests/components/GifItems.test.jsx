import { render, screen } from "@testing-library/react"
import { GifItems } from "../../src/Components/GifItems"

describe('Pruebas en GifItems', () => { 
    
    const title ='Saitama'
    const url = 'https://naruto.com/Naruto'
    
    
    test('Debe de hacer math con el snapshot', () => { 
        const { container } = render( <GifItems title={ title } url={ url }   /> );
        expect( container ).toMatchSnapshot();
    })

    test('Debe mostrar la imagen con el url y el ALT indicado ', () => { 
        render( <GifItems title={ title } url={ url } /> );
        // screen.debug();
        // expect( screen.getByRole( 'img' ).src ).toBe( url );
        // expect( screen.getByRole( 'img' ).alt ).toBe( title );
        const { src, alt } = screen.getByRole( 'img' );
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })

    test('Debe mostar un titulo en el componente ', () => {
        render( <GifItems title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();
    })
})






