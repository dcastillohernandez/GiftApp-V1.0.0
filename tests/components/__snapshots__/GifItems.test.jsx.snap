// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItems Debe de hacer math con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://naruto.com/Naruto"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
